xds/csds/csds_e2e_test.go

```diff
@@ -86,6 +86,34 @@ func Test(t *testing.T) {
 	grpctest.RunSubTests(t, s{})
 }
 
+// The following watcher implementations are no-ops since we don't really care
+// about the callback received by these watchers in the test. We only care
+// whether CSDS reports the expected state.
+
+type unimplementedListenerWatcher struct{}
+
+func (unimplementedListenerWatcher) OnUpdate(*xdsresource.ListenerResourceData) {}
+func (unimplementedListenerWatcher) OnError(error)                              {}
+func (unimplementedListenerWatcher) OnResourceDoesNotExist()                    {}
+
+type unimplementedRouteConfigWatcher struct{}
+
+func (unimplementedRouteConfigWatcher) OnUpdate(*xdsresource.RouteConfigResourceData) {}
+func (unimplementedRouteConfigWatcher) OnError(error)                                 {}
+func (unimplementedRouteConfigWatcher) OnResourceDoesNotExist()                       {}
+
+type unimplementedClusterWatcher struct{}
+
+func (unimplementedClusterWatcher) OnUpdate(*xdsresource.ClusterResourceData) {}
+func (unimplementedClusterWatcher) OnError(error)                             {}
+func (unimplementedClusterWatcher) OnResourceDoesNotExist()                   {}
+
+type unimplementedEndpointsWatcher struct{}
+
+func (unimplementedEndpointsWatcher) OnUpdate(*xdsresource.EndpointsResourceData) {}
+func (unimplementedEndpointsWatcher) OnError(error)                               {}
+func (unimplementedEndpointsWatcher) OnResourceDoesNotExist()                     {}
+
```

Comment:
I'm wondering if we should just have one `unimplementedGenericWatcher` struct; 1. since all of the implemented methods are no-ops 2. we instantiate an empty object when passing into WatchListener.

Thoughts?

Reply:
I agree. I had a similar thought when writing units for #5996. I ended up writing one a dummy delegation watcher there. 

LGTM. Ship it :)